### ISSUE TITLE
[ios] InteractionManager.runAfterInteractions doesn't run sometimes

### ISSUE THREAD

**Issue by Pyroboomka** [software-mansion/react-native-gesture-handler#1767]
## Description
Was doing some package updates in my application, so I bumped gesture-handler version to 2.1.0 from 1.10.3
Ran into some issues, where my existing code with InteractionManager stopped working correctly.
I used it mainly keep react-navigation transitions lag-free, so I defer fetching or render placeholder elements until transition ends with 
```
InterfactionManager.runAfterInteractions(() => {
  callBackend(); // or setIsReady(true);
})
```

The issue if using gesture-handler@2.0.0 it sometimes doesn't calls provided function on subsequent mounts of the screen.
I made a sample project and recorded a video to showcase the problem better: [https://youtu.be/fMUfxbykvHM](https://youtu.be/fMUfxbykvHM)

Not sure, maybe something changed in 2.0.0 with marking animations as interactions? Maybe issues are with animation interruption?

## Steps To Reproduce
Described above

### Expected behavior
runAfterInteractions callback fire every time.

### Actual behavior
runAfterInteractions callback does not gets called sometimes.

## Snack or minimal code example
Made a sample repo on ejected expo project - [https://github.com/Pyroboomka/ghinteractionmanager](https://github.com/Pyroboomka/ghinteractionmanager)
Relevant code is in navigation and screens folders.
Not sure that extra navigator is necessary or matter, but it's easier to demonstrate that it resets after stack unmount.
## Package versions
    "@react-navigation/native": "^6.0.2", // expo default, same behavior in 5.x.x versions
    "@react-navigation/stack": "^6.0.11", // same behavior in 5.x.x versions
    "react": "17.0.1",
    "react-native": "0.64.3", // same behavior in @0.66.4 .
    "react-native-screens": "~3.8.0", // does not matter I guess, enableScreens(false|true) doesn't change anything.
    "react-native-gesture-handler": "2.0.0",
    "react-native-reanimated": "~2.2.0" // does not matter, same issues in 2.3.0

**Comment by TaichiAkimoto** [comment]
I ran into this bug, too.
I've spent hours debugging until I found this issue...
so helpfull.

**Comment by j-piasecki** [comment]
Hi! Could you check if this patch solves the issue for you?
```diff
diff --git a/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx b/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
index f20afb5..4ce3b31 100755
--- a/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
+++ b/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
@@ -272,7 +272,7 @@ export default class Card extends React.Component<Props> {
     } = this.props;
 
     switch (nativeEvent.state) {
-      case GestureState.BEGAN:
+      case GestureState.ACTIVE:
         this.isSwiping.setValue(TRUE);
         this.handleStartInteraction();
         onGestureBegin?.();
```

**Comment by aleksandr-senichev** [comment]
@j-piasecki Nice, thank you! Works for `@react-navigation/stack: 5.14.9`

**Comment by Cattari** [comment]
@j-piasecki had the same issue on @react-navigation/stack: 6.0.11. Th fix is present in @react-navigation/stack: 6.3.17. Thank you!
